daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -601,12 +608,20 @@ object Speedy {
     def fromPureSExpr(
         compiledPackages: CompiledPackages,
         expr: SExpr,
+    ): Machine = fromPureAExpr(
+      compiledPackages = compiledPackages,
+      anf = flattenToAnf(expr),
+    )
+
+    def fromPureAExpr(
```

Comment:
De we really need one more builder for the Machine ? 
There is already quite a lot of parameters to those builders (and there will be some more to control output transaction versionning), and those parameters are quite sensitive to set up and have a subtle semantics. 
I am a bit worry that if there is too much builders people (where most of them are with default parameters) people will just overlook those parameters. 

Maybe we do not need the from `PureSExpr`  anymore.

Reply:
both are used, but I will have a look if one can be eliminated

Reply:
I had a look. I don't think we should remove either of these entry point (at the moment).
It would be tricky to remove the new `fromPureAExpr` because sometimes the caller has `AExpr` in their hands.
It would be trivial to remove/inline `fromPureSExpr`, but that would mean more place in the code which have to call `flattenToAnf`, and I think that would be worse.

Reply:
I'm fine with having another entry point for this.